compiler/src/dmd/func.d

```diff
@@ -2199,6 +2199,21 @@ extern (C++) class FuncDeclaration : Declaration
         }
         else
         {
+            foreach (v; closureVars)
+            {
+                v.doNotInferScope = true;
+                v.storage_class &= ~STC.maybescope;
+                if (v.isScope() || (v.storage_class & STC.scopeinferred))
```

Comment:
It shouldn't be possible to have `STC.scopeInferred` but not `STC.scope_`, did you encounter this?

Reply:
You're right, it is unnecessary. Fixed.